wp/lib/bap_wp/src/environment.ml

```diff
@@ -119,7 +122,7 @@ let wp_rec_call :
   (t -> Constr.t -> start:Graphs.Ir.Node.t -> Graphs.Ir.t -> t) ref =
   ref (fun _ _ ~start:_ _ -> assert false)
 
-let init_loop_unfold (num_unroll : int) : loop_handler =
+let init_loop_unfold (num_unroll : int) (ctx : Z3.context) : loop_handler =
```

Comment:
Not crazy about this `ctx` here, it seems unnecessary and hard to keep consistent with the `ctx` in `env`. Can't we just use the one in `env`?

Reply:
I totally forgot about the `ctx` in the `env`. Thanks for the catch.